src/User.php

```diff
@@ -1394,9 +1394,9 @@ public function post_updateItem($history = true)
         //handle timezone change for current user
         if ($this->fields['id'] == Session::getLoginUserID()) {
             if (null == $this->fields['timezone'] || 'null' === strtolower($this->fields['timezone'])) {
-                unset($_SESSION['glpi_tz']);
+                unset($_SESSION['glpitimezone']);
             } else {
-                $_SESSION['glpi_tz'] = $this->fields['timezone'];
+                $_SESSION['glpitimezone'] = $this->fields['timezone'];
             }
         }
```

Comment:
This whole block is useless now, the value will be inside `User::prepareInputForUpdate()` in the `foreach ($CFG_GLPI['user_pref_field'] as $f) {` loop.